undefined reference to `trunc'

Building locally results in a linker error referencing `trunc`.

    $ make
    build/output.o: In function `output_program':
    output.c:(.text+0x461): undefined reference to `trunc'
    collect2: error: ld returned 1 exit status

`trunc` is located in libm in my system, requiring  the addition of '-lm' in the linking phase.

I'm running glibc-2.20 and gcc 5.1.0 under Gentoo Linux.

--- a/Makefile
+++ b/Makefile
@@ -1,5 +1,5 @@
 CC=gcc
-CFLAGS=-pedantic -Wall -pedantic-errors -Wundef -Wstrict-prototypes  -Wmissing-declarations
+CFLAGS=-std=c99 -pedantic -Wall -pedantic-errors -Wundef -Wstrict-prototypes  -Wmissing-declarations
 
 all: tis
 
@@ -7,7 +7,7 @@ run: tis
 	./tis
 
 tis: src/tis.c build/program.o build/node.o build/util.o build/input_code.o build/node_list.o build/output.o
-	$(CC) $(CFLAGS) -o tis src/tis.c build/program.o build/node.o build/util.o build/input_code.o build/node_list.o build/output.o -lncurses
+	$(CC) $(CFLAGS) -o tis src/tis.c build/program.o build/node.o build/util.o build/input_code.o build/node_list.o build/output.o -lncurses -lm
 
 build/program.o: src/program.c src/program.h src/node.h src/node_list.h
 	$(CC) $(CFLAGS) -c src/program.c -o build/program.o

--- a/src/node.c
+++ b/src/node.c
@@ -36,7 +36,7 @@ Instruction *node_create_instruction(Node *n, Operation op) {
   return i;
 }
 
-void parse_location(const char *s, union Location *loc, LocationType *type) {
+static void parse_location(const char *s, union Location *loc, LocationType *type) {
   if (!s) { raise_error("no source was found"); }
 
   if (strcmp(s, "UP") == 0) {
@@ -69,7 +69,7 @@ void parse_location(const char *s, union Location *loc, LocationType *type) {
   }
 }
 
-void parse_mov(Node *n, const char *s) {
+static void parse_mov(Node *n, const char *s) {
   const int len = strlen(s+4);
   char *rem = (char *) malloc(sizeof(char) * len);
   strcpy(rem, s+4);
@@ -81,7 +81,7 @@ void parse_mov(Node *n, const char *s) {
   free(rem);
 }
 
-void parse_onearg(Node *n, InputCode *ic, const char *s, Operation op) {
+static void parse_onearg(Node *n, InputCode *ic, const char *s, Operation op) {
   const int len = strlen(s+4);
   char *rem = (char *) malloc(sizeof(char) * len);
   strcpy(rem, s+4);
@@ -159,6 +159,7 @@ void node_parse_line(Node *n, InputCode *ic, const char *s) {
 
   char ins[5];
   strncpy(ins, s, 3);
+  ins[3] = '\0';
 
   if (strcmp(ins, "MOV") == 0) {
     parse_mov(n, s);

--- a/src/program.c
+++ b/src/program.c
@@ -1,7 +1,9 @@
+#define _POSIX_C_SOURCE 200809L
 #include <stdio.h>
 #include <stdlib.h>
 #include <assert.h>
 #include <string.h>
+#include <unistd.h>
 
 #include "constants.h"
 #include "input_code.h"
@@ -9,7 +11,8 @@
 #include "util.h"
 #include "curses.h"
 
-Node *create_node(Program *p) {
+
+static Node *create_node(Program *p) {
   Node *n = (Node *) malloc(sizeof(Node));
   node_init(n);
   p->nodes = node_list_append(p->nodes, n);
@@ -61,7 +64,7 @@ int program_tick(const Program *p) {
   return all_blocked;
 }
 
-char *read_line(FILE *fp, char *line) {
+static char *read_line(FILE *fp, char *line) {
   size_t len = 0;
   ssize_t read = getline(&line, &len, fp);
 
@@ -73,7 +76,7 @@ char *read_line(FILE *fp, char *line) {
 }
 
 
-Node *create_input_node(Program *p, FILE *fp) {
+static Node *create_input_node(Program *p, FILE *fp) {
   Node *n = create_node(p);
 
   char *line = NULL;
@@ -101,7 +104,7 @@ Node *create_input_node(Program *p, FILE *fp) {
   return n;
 }
 
-Node *create_output_node(Program *p, FILE *fp) {
+static Node *create_output_node(Program *p, FILE *fp) {
   Node *n = create_node(p);
 
   char *line = NULL;

--- a/src/tis.c
+++ b/src/tis.c
@@ -8,7 +8,7 @@
 #include "output.h"
 
 
-int main() {
+int main(int argc, char **argv) {
   Program program;
   Program *p = &program;
 

